File: backend/src/domain/entities/MemberRole.ts
import { Entity, PrimaryColumn, CreateDateColumn } from 'typeorm';

@Entity('member_roles')
export class MemberRole {
  @PrimaryColumn({ name: 'member_id' })
  memberId: number;

  @PrimaryColumn({ name: 'role_id' })
  roleId: number;

  @CreateDateColumn({ type: 'timestamp', name: 'assigned_at' })
  assignedAt: Date;
}

File: backend_new_structure/common/src/main/java/com/service/common/entity/MemberRole.java
package com.service.common.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("member_roles")
public class MemberRole {

    @Column("member_id")
    private Long memberId;

    @Column("role_id")
    private Long roleId;

    @CreatedDate
    @Column("assigned_at")
    private LocalDateTime assignedAt;
}
